src/Cache/ShardedPhpRedis.php

```diff
@@ -0,0 +1,326 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\redis\Cache\PhpRedis.
+ */
+
+namespace Drupal\redis\Cache;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Cache\CacheTagsChecksumInterface;
+
+/**
+ * PhpRedis cache backend.
+ */
+class ShardedPhpRedis extends CacheBase {
```

Comment:
I need to read through this to fully understand it, but it's hard to see the differences compared to my implementation since the refactoring. I think a lot is actually very similar now, is there really reason now to keep both implementations?